### PR TITLE
Feat: add getAllSchools function

### PR DESCRIPTION
**Pull request by carissa-tang**
## Notion ticket link

[Get all schools](https://www.notion.so/uwblueprintexecs/Get-all-schools-f247ef4290bc4c6b8dbeb79ff603b71b)

## Implementation description
* Created `getAllSchools()` in the school service which retrieves all existing schools and returns an array of `SchoolResponseDTO`
* Updated both interface and implementation
* Wrote one test case (success)
* Moved test data into `/testUtils`

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/services/implementations/schoolService.ts [hunk 1]
```diff
@@ -18,6 +18,33 @@ class SchoolService implements ISchoolService {
     this.userService = userService;
   }
 
+  // This method gets all schools from the database.
+  async getAllSchools(): Promise<SchoolResponseDTO[]> {
```
Comment (mahadzaryab1) [inline backend/services/implementations/schoolService.ts]: nit: use documentation format starting with `/**`

File: backend/services/implementations/schoolService.ts [hunk 2]
```diff
@@ -18,6 +18,33 @@ class SchoolService implements ISchoolService {
...
+            id: school.id,
+            name: school.name,
+            country: school.country,
+            subRegion: school.subRegion,
+            city: school.city,
+            address: school.address,
+            teachers: teacherDTOs,
+          };
+        }),
+      );
```
Comment (mahadzaryab1) [inline backend/services/implementations/schoolService.ts]: thoughts on moving this logic to a helper function? I can see it being reused again in the future

File: backend/services/interfaces/schoolService.ts [hunk 3]
```diff
@@ -45,11 +45,19 @@ export interface SchoolResponseDTO {
  * on the School entity in the datbase.
  */
 export interface ISchoolService {
+  /**
+   * This method retrieve all Schools
```
Comment (mahadzaryab1) [inline backend/services/interfaces/schoolService.ts]: retrieves**

File: backend/services/implementations/__tests__/schoolService.test.ts [hunk 4]
```diff
@@ -31,72 +38,48 @@ describe("mongo schoolService", (): void => {
...
-        role: "Admin",
-      },
-      {
-        id: "56cb91bdc3464f14678934cb",
-        firstName: "Teacher",
-        lastName: "Two",
-        authId: "456",
-        role: "Admin",
-      },
-    ];
```
Comment (mahadzaryab1) [inline backend/services/implementations/__tests__/schoolService.test.ts:53]: thanks for refactoring this!

**Review by mahadzaryab1 — COMMENTED** [review]
great work - left a couple of comments

File: backend/services/implementations/schoolService.ts (hunk 1 above)
Comment (carissa-tang) [reply backend/services/implementations/schoolService.ts]: just wondering - why do we add documentation to both the interface and implementation? is that a common practice you've seen?

File: backend/services/implementations/schoolService.ts (hunk 2 above)
Comment (carissa-tang) [reply backend/services/implementations/schoolService.ts]: good idea :)

File: backend/services/implementations/schoolService.ts (hunk 1 above)
Comment (mahadzaryab1) [reply backend/services/implementations/schoolService.ts]: we can just add it to the interface - you don't have to do add it to the implementation but you can if you want to!

File: backend/services/implementations/__tests__/schoolService.test.ts [hunk 5]
```diff
@@ -31,72 +38,48 @@ describe("mongo schoolService", (): void => {
...
+    const res = await schoolService.getAllSchools();
+
+    res.forEach((school: SchoolResponseDTO, i) => {
+      expect(school.id).not.toBeNull();
+      expect(school.name).toEqual(testSchools[i].name);
+      expect(school.country).toEqual(testSchools[i].country);
+      expect(school.subRegion).toEqual(testSchools[i].subRegion);
+      expect(school.city).toEqual(testSchools[i].city);
+      expect(school.address).toEqual(testSchools[i].address);
+      expect(school.teachers).toEqual(testUsers);
```
Comment (mahadzaryab1) [inline backend/services/implementations/__tests__/schoolService.test.ts]: thoughts on moving this into a separate util function as well? then we can simplify this code to something like 
```typescript
res.forEach((school: SchoolResponseDTO, i) => {
  assertResponseMatchesExpected(school, testSchools[i])
}
```

File: backend/services/implementations/__tests__/schoolService.test.ts [hunk 6]
```diff
@@ -31,72 +38,48 @@ describe("mongo schoolService", (): void => {
...
-    expect(createdSchool.name).toEqual(testSchool.name);
-    expect(createdSchool.country).toEqual(testSchool.country);
-    expect(createdSchool.subRegion).toEqual(testSchool.subRegion);
-    expect(createdSchool.city).toEqual(testSchool.city);
-    expect(createdSchool.address).toEqual(testSchool.address);
+    expect(createdSchool.name).toEqual(testSchools[0].name);
+    expect(createdSchool.country).toEqual(testSchools[0].country);
+    expect(createdSchool.subRegion).toEqual(testSchools[0].subRegion);
+    expect(createdSchool.city).toEqual(testSchools[0].city);
+    expect(createdSchool.address).toEqual(testSchools[0].address);
```
Comment (mahadzaryab1) [inline backend/services/implementations/__tests__/schoolService.test.ts]: if you decide to do the above refactor - you can replace this entire block with that function call

**Review by mahadzaryab1 — COMMENTED** [review]
couple more comments

**Review by mahadzaryab1 — APPROVED** [review]
lgtm - nicely done!

File: backend/services/interfaces/schoolService.ts [hunk 7]
```diff
@@ -45,11 +45,19 @@ export interface SchoolResponseDTO {
  * on the School entity in the datbase.
  */
 export interface ISchoolService {
+  /**
+   * This method retrieves all Schools
+   * @param
+   * @returns returns array of SchoolResponseDTO
+   * @throws Error if retrieval fails
+   */
```
Comment (joyce-shi) [inline backend/services/interfaces/schoolService.ts:53]: cute docs 🤙

**Review by joyce-shi — APPROVED** [review]
LGTM! 🛳

**Pull request merged**